### ISSUE TITLE
[2.5.1] Release test workspace_template_batch_inference.aws failed

### ISSUE THREAD

**Issue by can-anyscale** [ray-project/ray#36505]
### What happened + What you expected to happen

Release test workspace_template_batch_inference.aws failed

Buildkite:
- https://buildkite.com/ray-project/release-tests-branch/builds/1802#0188c475-be45-4b7a-9cef-5403b6827f37

Error:

`ImportError: cannot import name 'deprecate_ndim_indexing' from 'pandas.core.indexers' (/home/ray/anaconda3/lib/python3.9/site-packages/pandas/core/indexers/__init__.py)`

### Versions / Dependencies

2.5.1

### Reproduction script

buildkite

### Issue Severity

High: It blocks me from completing my task.

**Comment by justinvyu** [comment]
@can-anyscale 

For some reason, on [the latest run that SUCCEEDED](https://buildkite.com/ray-project/release-tests-branch/builds/1792#0188b33a-905e-4ab7-a1cc-e2b98e8b2656), the pandas version for `ray-ml/nightly` is `pandas==1.5.3`.

On the linked run that FAILED, the pandas version is now pinned to `pandas==1.3.5`. Any idea what changed here? I don't see an explicit pin on pandas anywhere, so maybe it's caused by a different dependency getting updated?

**Comment by can-anyscale** [comment]
It would be very hard to track down which package cause pandas to change its version. My best guess is due to some discrepancy (like some packages are pinned under different versions) between ray nightly wheel and ray 2.5.1 wheel.

**Comment by krfricke** [comment]
TLDR: Should be fixed, will kick off again today.


This was probably due to our change to Python 3.8:

- In https://github.com/ray-project/ray/commit/7d3bc5e7a524aa6cefaf39c8ad7529ec09a89c3d we move to Python 3.8
- We had to pin modin to 0.12.x (and hence pandas to 1.3.5) to avoid including too many code changes in the CI switch
- This affected the `ray-ml` Docker images on nightly for a day or so. Previously Python 3.8 installed a later modin, but we downgraded this so that the CI would pass
- We then re-upgraded modin and pandas here https://github.com/ray-project/ray/pull/36451

The main issue is that we're using the nightly ray-ml base images. This should be fixed by BYOD in the future.

**Comment by krfricke** [comment]
Confirmed that 2.5.1 images have the correct pandas versions installed, and the test is [passing on master](https://buildkite.com/ray-project/release-tests-branch/builds/1816#0188dcd4-f04e-4a68-be57-3ac9f0fb8979). I'll close this issue.
